client/my-sites/plan-features-2023-grid/index.tsx

```diff
@@ -242,9 +245,19 @@ export class PlanFeatures2023Grid extends Component<
 	PlanFeatures2023GridType,
 	PlanFeatures2023GridState
 > {
-	state = {
-		showPlansComparisonGrid: false,
-	};
+	constructor( props: PlanFeatures2023GridType ) {
+		super( props );
+
+		this.state = {
+			showPlansComparisonGrid: false,
+			selectedStorage: {},
+		};
+
+		this.props.planProperties.forEach( ( planProperties ) => {
+			const { planName } = planProperties;
+			this.state.selectedStorage[ planName ] = '';
```

Comment:
and why are we initializing these to an empty string?

Reply:
I'm trying to track storage add-ons on a per plan basis. Typescript was unhappy that state wasn't initialized without the correct plan keys.

Totally open to suggestions here though -- I might be missing some understanding on the typescript side of things

Reply:
It looks a little suspicious to me that we have to do this. I need to take a closer look. Have you tried fiddling with optional operators `foo[ bar ]?.baz`, etc?

Reply:
> Have you tried fiddling with optional operators foo[ bar ]?.baz, etc?

I did some experimentation with some checks like this but nothing detailed 🤔 

I'll spend some time digging into this. Thanks for all the help btw @chriskmnds!

Reply:
https://github.com/Automattic/wp-calypso/pull/79041#discussion_r1258936928